### PR TITLE
DAO_09

### PR DESCRIPTION
**Pull request by engineer027**
1. create a new package:
cars;
drivers;
manufacturers.
2. Sorted all my Controllers.

### CODE REVIEW

File: src/main/java/mate/academy/controllers/cars/AddDriverToCarController.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package mate.academy.controllers;
+package mate.academy.controllers.cars;
```
Comment (Ketch92) [inline src/main/java/mate/academy/controllers/cars/AddDriverToCarController.java]: the packages should have a singular naming "controllers -> controller", cars -> cars

File: src/main/java/mate/academy/controllers/drivers/DeleteDriverController.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package mate.academy.controllers;
+package mate.academy.controllers.drivers;
```
Comment (Ketch92) [inline src/main/java/mate/academy/controllers/drivers/DeleteDriverController.java]: the same

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Dao;
+import mate.academy.lib.Inject;
+import mate.academy.model.Driver;
+import mate.academy.service.DriverService;
+
+@Dao
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
```
Comment (Ketch92) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: I'm not sure if we need to inject a service into another service. I think there should be DriverDao. just FYI, I don't demand this change

File: src/main/java/mate/academy/service/DriverService.java [hunk 4]
```diff
@@ -13,4 +14,6 @@
     Driver update(Driver driver);
 
     boolean delete(Long id);
+
+    Optional<Driver> findByLogin(String login);
```
Comment (Ketch92) [inline src/main/java/mate/academy/service/DriverService.java]: If I'm right and we should inject the dao to AuthenticationService then service should return Driver and not Optional of Driver

File: src/main/java/mate/academy/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        if (url.equals("/login") || url.equals("/drivers/create")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
```
Comment (Ketch92) [inline src/main/java/mate/academy/web/filters/AuthenticationFilter.java]: check common mistakes, you can do it with single if construction

**Review by Ketch92 — CHANGES_REQUESTED** [review]

**Review by Ketch92 — APPROVED** [review]

File: src/main/java/mate/academy/model/Driver.java [hunk 6]
```diff
@@ -36,6 +46,14 @@ public String getLicenceNumber() {
...
+    public String getLogin() {
+        return login;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
     @Override
     public boolean equals(Object o) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/model/Driver.java:58]: let's add login and password to equals and hashcode

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import mate.academy.model.Driver;
+
+@Dao
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driverFromDB = driverDao.findByLogin(login)
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: Read about it in the Slack)

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/security/AuthenticationService.java [hunk 8]
```diff
@@ -1,8 +1,9 @@
 package mate.academy.security;
 
+import java.util.Optional;
 import mate.academy.exception.AuthenticationException;
 import mate.academy.model.Driver;
 
 public interface AuthenticationService {
-    Driver login(String login, String password) throws AuthenticationException;
+    Optional<Driver> login(String login, String password) throws AuthenticationException;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/AuthenticationService.java]:
```suggestion
    Driver login(String login, String password) throws AuthenticationException;
```

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 9]
```diff
@@ -12,11 +13,9 @@
...
     @Override
-    public Driver login(String login, String password) throws AuthenticationException {
-        Driver driverFromDB = driverDao.findByLogin(login)
-                .orElseThrow(() -> new AuthenticationException("Incorrect login or password"));
-
-        if (driverFromDB.getPassword().equals(password)) {
+    public Optional<Driver> login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driverFromDB = driverDao.findByLogin(login);
+        if (driverFromDB.isPresent() && driverFromDB.get().getPassword().equals(password)) {
             return driverFromDB;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]:
```suggestion
    public Driver login(String login, String password) throws AuthenticationException {
        Optional<Driver> driverFromDB = driverDao.findByLogin(login);
        if (driverFromDB.isPresent() && driverFromDB.get().getPassword().equals(password)) {
            return driverFromDB.get();
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**